salt/grains/core.py

```diff
@@ -1668,6 +1668,28 @@ def append_domain():
         grain['append_domain'] = __opts__['append_domain']
     return grain
 
+def fqdns():
+    '''
+    Return all FQDNs grains for the system
+    '''
+    # Provides:
+    # fqdns
+
+    grains = {}
+    fqdns = set()
+
+    addresses = sum([[_INTERFACES[k]['inet'][0]['address'], 
+        _INTERFACES[k]['inet6'][0]['address']] 
+        for k in _INTERFACES if k != 'lo'], [])
+
+    for ip_addr in addresses:
+        try:
+            fqdns.add(socket.gethostbyaddr(ip_addr)[0])
+        except (socket.error, socket.herror, socket.gaierror, socket.timeout):
```

Comment:
You should not make the exception invisible. In case of exception please at least produce a log output.